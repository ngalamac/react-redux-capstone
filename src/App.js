import {
  createBrowserRouter,
  createRoutesFromElements,
  Route,
  RouterProvider,
} from 'react-router-dom';

import Home from './components/pages/Home';
import Details from './components/pages/Details';

const router = createBrowserRouter(
  createRoutesFromElements(
    <>
      <Route path="/" element={<Home />} />
      <Route path="/:code" element={<Details />} />
    </>,
  ),
);
function App() {
  return <RouterProvider router={router} />;
}

export default App;
