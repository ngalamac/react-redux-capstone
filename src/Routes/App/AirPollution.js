import React from 'react';
import { Outlet } from 'react-router-dom';
import Nav from '../../Components/Nav/Nav';

const AirPollution = () => (
  <>
    <Nav />
    <Outlet />
  </>
);

export default AirPollution;
