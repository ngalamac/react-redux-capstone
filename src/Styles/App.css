@import url('./reset.css');
@import url('./variables.css');
@import url('./base.css');
@import url('./button.css');
@import url('../Components/Nav/nav.css');
@import url('../Routes/Home/home.css');
@import url('../Components/Hero/hero.css');
@import url('../Components/Continents/continents.css');
@import url('../Components/Continent/continent.css');
@import url('../Routes/Countries/countries.css');
@import url('../Components/Country/country.css');
@import url('../Routes/Pollution/pollution.css');
@import url('../Components/Pollution/pollutionDetail.css');
@import url('../Components/HeroContinent/heroContinent.css');
@import url('../Components/SearchCountry/searchCountry.css');
@import url('./loader.css')
