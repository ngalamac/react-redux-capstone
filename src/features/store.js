import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './countries/countrySlice';

const store = configureStore({
  reducer: {
    country: countryReducer,
  },
});

export default store;
