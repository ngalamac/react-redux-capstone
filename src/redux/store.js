import { configureStore } from '@reduxjs/toolkit';
import teamsReducer from './footballSlice';

const store = configureStore({
  reducer: {
    teams: teamsReducer,
  },
});

export default store;
