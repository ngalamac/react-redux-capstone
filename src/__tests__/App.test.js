import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../features/store';
import App from '../App';

afterEach(cleanup);

describe('App', () => {
  it('App to match Snapshot', () => {
    const app = render(<Provider store={store}><App /></Provider>);
    expect(app).toMatchSnapshot();
  });
});
