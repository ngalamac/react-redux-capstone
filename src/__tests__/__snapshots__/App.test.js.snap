// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App to match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="homeMainContainer"
      >
        <header>
          <nav
            class="navbar"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <button
                class="button"
                type="button"
              >
                <svg
                  class="icons"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
            </a>
            <h2>
              Country List
            </h2>
            <div
              class="audioSettingDiv"
            >
              <svg
                class="icons margin-right"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 352 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
                />
              </svg>
              <svg
                class="icons"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
                />
              </svg>
            </div>
          </nav>
        </header>
        <div
          class="homeBg"
        >
          <h1
            class="homeTitle"
          >
             Nation Navigator
          </h1>
        </div>
        <div
          class="search"
        >
          <select
            class="select"
            name="region"
            type="text"
          >
            <option
              value=""
            >
              View by Region
            </option>
            <option
              value="Asia"
            >
              Asia Pacific
            </option>
            <option
              value="America"
            >
              America
            </option>
            <option
              value="Africa"
            >
              Africa
            </option>
            <option
              value="Europe"
            >
              Europe
            </option>
            <option
              value="Oceania"
            >
              Oceania
            </option>
          </select>
        </div>
        <h1>
          Loading...
        </h1>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="homeMainContainer"
    >
      <header>
        <nav
          class="navbar"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <button
              class="button"
              type="button"
            >
              <svg
                class="icons"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                />
              </svg>
            </button>
          </a>
          <h2>
            Country List
          </h2>
          <div
            class="audioSettingDiv"
          >
            <svg
              class="icons margin-right"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
              />
            </svg>
            <svg
              class="icons"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
              />
            </svg>
          </div>
        </nav>
      </header>
      <div
        class="homeBg"
      >
        <h1
          class="homeTitle"
        >
           Nation Navigator
        </h1>
      </div>
      <div
        class="search"
      >
        <select
          class="select"
          name="region"
          type="text"
        >
          <option
            value=""
          >
            View by Region
          </option>
          <option
            value="Asia"
          >
            Asia Pacific
          </option>
          <option
            value="America"
          >
            America
          </option>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
        </select>
      </div>
      <h1>
        Loading...
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
